src/Microsoft.ApiDesignGuidelines.Analyzers/Core/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -15,25 +16,63 @@ public abstract class CollectionsShouldImplementGenericInterfaceAnalyzer : Diagn
-        
+
         private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.CollectionsShouldImplementGenericInterfaceMessage), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.CollectionsShouldImplementGenericInterfaceDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
-        
+
         internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182132.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(
+               (context) =>
+               {
+                   var collectionType = WellKnownTypes.ICollection(context.Compilation);
+                   var genericCollectionType = WellKnownTypes.GenericICollection(context.Compilation);
+                   var enumerableType = WellKnownTypes.IEnumerable(context.Compilation);
+                   var genericEnumerableType = WellKnownTypes.GenericIEnumerable(context.Compilation);
+                   var listType = WellKnownTypes.IList(context.Compilation);
+                   var genericListType = WellKnownTypes.GenericIList(context.Compilation);
+
+                   if (collectionType == null || genericCollectionType ==null ||
```

Comment:
Even if any one of these types is missing, we would not run the rule? Should these be and'ed instead?